### PR TITLE
User-facing Balancer-specific error codes

### PR DESCRIPTION
**Pull request by PaulMcInnis**
Now we process balancer-specific (BAL#) errors when thrown via revert or simulation in swap & pool create

To make it more human-readable, i've _generated_ error descriptions from the codes themselves using the balancer codebase as context. If we prefer we can keep the more basic `XXX_XXX` type codes. If we keep the generated ones, we might consider pulling them into `balanceError` in `useAddLiquidity`. 

_edit: for codes which we have documentation i've updated those strings._

swap:
image

pool create (I've since added BAL# to the end of these strings):
HONEY STGUSDC
The amplification parameter is below the minimum allowed
Swap fee percentage is below the minimum allowed

NOTE: 
- you can test the simulation errors by overriding swap amounts / rate/ fee with 0n in `useCreatePool`.
- ~~you can test the revert errors by forcing a tx by making this change the CreatePageContent.tsx L562~~ this doesn't revert the same way
    ```
          writeCreatePool={() => {
            console.log("createPoolArgs", createPoolArgs);
            const data = encodeFunctionData({
              abi: createPoolArgs.abi,
              functionName: createPoolArgs.functionName,
              args: createPoolArgs.args,
            });
            writeCreatePool({
              address: createPoolArgs.address,
              data,
              value: createPoolArgs.value,
            });
          }}
    ```
- you can test on swap by just attempting to swap 1 BERA:HONEY on bartio (at the time of writing)

### CODE REVIEW

File: packages/berajs/src/utils/errorMessages.ts [hunk 1]
```diff
@@ -44,6 +44,344 @@ const errorMsgMap: ErrorMessages = {
       errorMSG: "You rejected the transaction.",
     },
   },
+  // NOTE: these codes are from https://github.com/balancer/balancer-v2-monorepo/blob/master/pkg/balancer-js/src/utils/errors.ts
+  BALANCER: {
+    BAL000: {
+      keywords: ["BAL#000"],
+      errorMSG: "Addition overflow occurred during calculation.",
```
Comment (PaulMcInnis) [inline packages/berajs/src/utils/errorMessages.ts:51]: open to including the BAL# in the message

**Review by bearpong — CHANGES_REQUESTED** [review]
Nice use of the errorMessages.ts though I think we should not fallback to a default message if the BAL error is not mapped in the array. I would match the BAL#000 error, and if present map it and always show it in the UI. It'd be easier to debug user issues when reported.

After that please do the same on useAddLiquidity as you suggested

File: packages/berajs/src/utils/errorMessages.ts (hunk 1 above)
Comment (bearpong) [reply packages/berajs/src/utils/errorMessages.ts]: yes please

**Comment by PaulMcInnis** [comment]
OK added this to useAddLiquidity 👍

**Comment by PaulMcInnis** [comment]
OK after reviewing https://docs.balancer.fi/reference/contracts/error-codes.html I've pulled in a bunch more codes

**Comment by PaulMcInnis** [comment]
OK I will fix handling for swap paths shortly

**Comment by PaulMcInnis** [comment]
Added coverage for this:
image

here's an existing error as well
image

**Comment by PaulMcInnis** [comment]
OK I will add handling for tx timeout due to swap deadline next

**Comment by PaulMcInnis** [comment]
image
already covered

**Review by bearpong — APPROVED** [review]

**Pull request merged**